lib/Index/Index.cpp

```diff
@@ -825,10 +825,10 @@ static bool isTestCandidate(ValueDecl *D) {
 
   // 5. ...is of at least 'internal' accessibility (unless we can use
   //    Objective-C reflection)...
-#if SWIFT_OBJC_INTEROP
-  if (D->getFormalAccess() < Accessibility::Internal)
+  if (!D->getASTContext().LangOpts.EnableObjCInterop &&
+      (D->getFormalAccess() < Accessibility::Internal ||
+      parentNTD->getFormalAccess() < Accessibility::Internal))
```

Comment:
Is this change to also checking the parent's formal access covered by the tests? It wasn't clear to me.